### PR TITLE
pass config to vite-plugin-svelte instead of reloading it

### LINKED ISSUE #4909: Gracefully recover from malformed svelte.config.js during development
### Describe the problem

At present if I make changes to `svelte.config.js`, the during development, the DEV server tries to load it and crashes because I've set my editor to autosave and have not yet completed typing. This is frustrating, especially when working on a Vite plugin or configuring other build scripts that require frequent changes.

### Describe the proposed solution

If the config load succeeded before, reuse the old config and a present a warning about the malformed `svelte.config.js` in the terminal. I assume it's as simple as a adding a try-catch somewhere, though I may be wrong.

### Alternatives considered

_No response_

### Importance

would make my life easier

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by Rich-Harris**
At present, `svelte.config.js` is loaded multiple times — once by SvelteKit, then again by vite-plugin-svelte (twice, during `svelte-kit build`). Aside from being marginally less efficient than reusing the config, this can result in surprising behaviour if you're doing any kind of setup work in the config file.

This uses the new `configFile: false` option and passes config directly to the plugin instead.

It also fixes #4909.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpx changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

File: packages/kit/src/core/build/build_server.js [hunk 1]
```diff
@@ -232,9 +232,12 @@ export async function build_server(
...
 			svelte({
 				extensions: config.extensions,
+				preprocess: config.preprocess,
 				compilerOptions: {
+					...config.compilerOptions,
 					hydratable: !!config.kit.browser.hydrate
-				}
+				},
+				configFile: false
 			})
```
[the remark this answers is not in the document]
Comment (dominikg) [reply packages/kit/src/core/build/build_server.js:241]: yeah, sveltekit config validation fortunately allows unknown keys and copies them along, the above should work. 

Even when we move the vite plugin options to their own key, the config file is the only place for users to customize vite-plugin-svelte in kit, unless we kind of duplicate the options inside the kit namespace or seal it off and only pass through a subset

File: packages/kit/src/core/dev/index.js [hunk 2]
```diff
@@ -60,15 +60,18 @@ export async function dev({ cwd, port, host, https, config }) {
 		plugins: [
 			svelte({
 				extensions: config.extensions,
+				preprocess: config.preprocess,
```
Comment (dominikg) [inline packages/kit/src/core/dev/index.js:63]:
```suggestion
				...config,
```

File: packages/kit/src/core/build/build_server.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (dominikg) [reply packages/kit/src/core/build/build_server.js:241]: damnit, v-p-s is going to complain about the inline `kit` key. need to add that to the allowed list.

**Comment by Rich-Harris** [comment]
this should be good to merge. it now gracefully restarts the entire dev server on config changes

File: packages/kit/package.json [hunk 3]
```diff
@@ -10,7 +10,7 @@
 	"homepage": "https://kit.svelte.dev",
 	"type": "module",
 	"dependencies": {
-		"@sveltejs/vite-plugin-svelte": "^1.0.0-next.32",
+		"@sveltejs/vite-plugin-svelte": "1.0.0-next.44",
```
Comment (dominikg) [inline packages/kit/package.json:13]: is the fixed version intentional?

Comment (Rich-Harris) [reply packages/kit/package.json:13]: nope!

```suggestion
		"@sveltejs/vite-plugin-svelte": "^1.0.0-next.44",
```

**Review by bluwy — APPROVED** [review]

**Pull request merged**